scripts/build/termux_step_get_dependencies.sh

```diff
@@ -82,28 +63,20 @@ termux_step_get_dependencies() {
 					fi
 				)
 			fi
-			mkdir -p $TERMUX_BUILT_PACKAGES_DIRECTORY
+			mkdir -p "$TERMUX_BUILT_PACKAGES_DIRECTORY"
 			if [[ "$cyclic_dependence" == "false" && ( "$TERMUX_WITHOUT_DEPVERSION_BINDING" == "false" || "$TERMUX_ON_DEVICE_BUILD" == "false" ) ]]; then
 				echo "$DEP_VERSION" > "$TERMUX_BUILT_PACKAGES_DIRECTORY/$PKG"
 			fi
-		else
-		# Build dependencies
-			if [[ -z "$PKG" ]]; then
-				continue
-			elif [[ "$PKG" == "ERROR" ]]; then
-				termux_error_exit "Obtaining buildorder failed"
-			fi
+		else # Build dependencies
 			# Built dependencies are put in the default TERMUX_OUTPUT_DIR instead of the specified one
 			if [[ "$TERMUX_FORCE_BUILD_DEPENDENCIES" == "true" ]]; then
-				[[ "$TERMUX_QUIET_BUILD" != true ]] && echo "Force building dependency $PKG..."
-				if [[ "$TERMUX_ON_DEVICE_BUILD" == "true" ]] && ! termux_package__is_package_on_device_build_supported "$PKG_DIR"; then
-					echo "Building $PKG on device is not supported. Consider passing -I flag to download it instead"
-					return 1
-				fi
-				read DEP_ARCH DEP_VERSION DEP_VERSION_PAC < <(termux_extract_dep_info $PKG "${PKG_DIR}")
+				[[ "$TERMUX_QUIET_BUILD" != "true" ]] && echo "Force building dependency $PKG..."
+				read DEP_ARCH DEP_VERSION DEP_VERSION_PAC DEP_ON_DEVICE_NOT_SUPPORTED < <(termux_extract_dep_info $PKG "${PKG_DIR}")
+				[[ "$TERMUX_ON_DEVICE_BUILD" == "true" && "$DEP_ON_DEVICE_NOT_SUPPORTED" == "true" ]] \
```

Comment:
As above, I think a bracegroup would improve readability here.
```suggestion
				[[ "$TERMUX_ON_DEVICE_BUILD" == "true" && "$DEP_ON_DEVICE_NOT_SUPPORTED" == "true" ]] && {
```

Reply:
Same as above, the whole point is to remove if-then-else block from here.